inc/health/checks/class-filesystemcheck.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Pressbooks\Health\Checks;
+
+use Pressbooks\Health\Check;
+use Pressbooks\Health\Result;
+use Symfony\Component\Process\Process;
+
+class FilesystemCheck extends Check {
+	public function run(): Result {
+		$result = Result::make();
+
+		if ( ! $this->canConnectToFilesystem() ) {
+			return $result->failed( 'Failed to obtain filesystem write access.' );
+		}
+
+		if ( ! $this->canWriteToFilesystem() ) {
+			return $result->failed( 'The filesystem is not writable.' );
+		}
+
+		if ( ! $this->canReadFromFilesystem() ) {
+			return $result->failed( 'The filesystem is not readable.' );
+		}
+
+		$disk_usage = $this->getDiskUsagePercentage();
+
+		// TODO: allow users to customise the failure threshold
+		if ( $disk_usage > 90 ) {
+			return $result->failed( "The disk is almost full ({$disk_usage}% used)." );
+		}
+
+		return $result->ok();
+	}
+
+	protected function canConnectToFilesystem(): bool {
+		global $wp_filesystem;
+
+		if ( ! WP_Filesystem() ) {
+			return false;
+		}
+
+		return $wp_filesystem->connect();
```

Comment:
Cara, does it sounds better? 
```php
global $wp_filesystem;
return WP_Filesystem() ? $wp_filesystem->connect() : false;
```
I think this reduction still sounds clear and simpler to me.

Reply:
Yep, will do